theseus/src/event/mod.rs

```diff
@@ -132,6 +145,11 @@ impl Drop for LoadingBarId {
                 #[cfg(not(any(feature = "tauri", feature = "cli")))]
                 bars.remove(&loader_uuid);
             }
+            let _ = SafeProcesses::complete(
+                crate::state::ProcessType::LoadingBar,
+                loader_uuid,
+            )
+            .await;
```

Comment:
Basically a state-stored collection of uuids representing processes that we do not want to interrupt by the app's closing.
You add it to safeprocesses, then the app will check if there are any running proccesses of this nature before it allows the app to be exited. (ie: "Are you sure you want to close? You have a download in progress"). you can then mark the process as complete to lose it from the list

This goes pretty hand in hand with the loading bars where almost all the loading bar operations also create the safe processes as well.

Reply:
That being said since I've now removed the settings sync part, and theres only loading bars for now, we could coalesque it into the loading bar code exclusively if you think there are no other kinds of processes in the future that we would want a warning before exiting.